clarity/evaluator/haspi/eb.py

```diff
@@ -1457,176 +1792,247 @@ def bm_covary(xBM, yBM, segsize, fsamp):
+    is a modification of Tan et al.[1]_ . The cross-covariance is also output
+    with a frequency weighting that reflects the loss of IHC synchronization at high
+    frequencies Johnson[2]_.
 
     Arguments:
-        sigcov (np.array) : [nchan,nseg] of cross-covariance values
-        sigMSx (np.array) : [nchan,nseg] of reference signal MS values
-        thr : threshold in dB SL to include segment ave over freq in average
+    signal_cross_covariance (np.array): [nchan,nseg] of cross-covariance values
+    reference_signal_mean_square (np.array): [nchan,nseg] of reference signal MS
+        values
+    threshold_db (): threshold in dB SL to include segment ave over freq in
+        average
+    lp_filter (list): LP filter order
+    freq_cutoff (list): Cutoff frequencies in Hz
 
     Returns:
-        avecov : cross-covariance in segments averaged over time and frequency
-        syncov : cross-coraviance array, 6 different weightings for loss of
+    average_covariance (): cross-covariance in segments averaged over time and
+        frequency
+    ihc_sync_covariance (): cross-coraviance array, 6 different weightings for loss of
               IHC synchronization at high frequencies:
               LP Filter Order     Cutoff Freq, kHz
                 1              1.5
                 3              2.0
                 5              2.5, 3.0, 3.5, 4.0
 
+    References:
+
+    .. [1] Tan CT, Moore, BCJ, Zacharov N, Mattila VV (2004) Predicting the Perceived
+           Quality of Nonlinearly Distorted Music and Speech Signals. J Audio Eng Soc
+           52(9):900-914. Available at.
+           <http://www.aes.org/e-lib/browse.cfm?elib=13013>.
+
+    .. [2] Johnson DH (1980) The relationship between spike rate and synchrony in
+           responses of auditory‐nerve fibers to single tones J Acoustocal Soc of Am
+           68:1115 Available at.
+           <https://doi.org/10.1121/1.384982>
+    Updates:
     James M. Kates, 28 August 2012.
     Adjusted for BM vibration in dB SL, 30 October 2012.
     Threshold for including time-freq tile modified, 30 January 2013.
     Version for different sync loss, 15 February 2013.
-
     Translated from MATLAB to Python by Gerardo Roa Dabike, September 2022.
     """
 
     # Array dimensions
-    nchan = sigcov.shape[0]
+    n_channels = signal_cross_covariance.shape[0]
 
     # Initialize the LP filter for loss of IHC synchronization
-    cfreq = CenterFreq(nchan)  # Center frequencies in Hz on an ERB scale
-    p = np.array([1, 3, 5, 5, 5, 5])  # LP filter order
-    fcut = 1000 * np.array([1.5, 2.0, 2.5, 3.0, 3.5, 4.0])  # Cutoff frequencies in Hz
-    fsync = np.zeros((6, nchan))  # Array of filter freq resp vs band center freq
+    _center_freq = center_frequency(
+        n_channels
+    )  # Center frequencies in Hz on an ERB scale
+    if lp_filter_order is None:
+        lp_filter_order = np.array([1, 3, 5, 5, 5, 5])  # LP filter order
+    if freq_cutoff is None:
```

Comment:
Would move comment to before line for better wrapping.  The frequencies are in 'kHz' not 'Hz'